tesseract_collision/include/tesseract_collision/core/collision_margin_data.h

```diff
@@ -0,0 +1,227 @@
+/**
+ * @file collision_margin_data.h
+ * @brief This is used to store collision margin information
+ *
+ * It should be used to perform continuous contact checking.
+ *
+ * @author Levi Armstrong
+ * @date January 18, 2018
+ * @version TODO
+ * @bug No known bugs
+ *
+ * @copyright Copyright (c) 2018, Southwest Research Institute
+ *
+ * @par License
+ * Software License Agreement (Apache License)
+ * @par
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * @par
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef TESSERACT_COLLISION_COLLISION_MARGIN_DATA_H
+#define TESSERACT_COLLISION_COLLISION_MARGIN_DATA_H
+
+#include <tesseract_common/macros.h>
+TESSERACT_COMMON_IGNORE_WARNINGS_PUSH
+#include <Eigen/Core>
+#include <string>
+#include <unordered_map>
+TESSERACT_COMMON_IGNORE_WARNINGS_POP
+
+#include <tesseract_common/types.h>
+
+namespace tesseract_collision
+{
+/** @brief Identifies how the provided contact margin data should be applied */
+enum class CollisionMarginOverrideType
+{
+  /** @brief Do not apply contact margin data */
+  NONE,
+  /** @brief Relace the contact managers */
+  REPLACE,
+  /** @brief Override the contact managers default margin only */
+  OVERRIDE_DEFAULT_MARGIN,
+  /** @brief Override the contact managers pair margin only. This does not preserve any existing pair margin data */
+  OVERRIDE_PAIR_MARGIN,
+  /** @brief Modify the contact managers pair margin only and preserves existing pair margin data */
```

Comment:
You might explicitly state here that duplicate pairs will not be modified.

Reply:
Ah, so duplicate will be modified. I will update the documentation.